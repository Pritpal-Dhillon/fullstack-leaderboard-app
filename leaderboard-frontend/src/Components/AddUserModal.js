import React from "react";
import { Button, Modal, Form } from "react-bootstrap";

const AddUserModal = ({ show, handleClose, addUser, newUser, setNewUser }) => (
  <Modal show={show} onHide={handleClose} centered>
    <Modal.Header closeButton>
      <Modal.Title>Add New User</Modal.Title>
    </Modal.Header>
    <Modal.Body>
      <Form.Group>
        <Form.Label>Name</Form.Label>
        <Form.Control
          type="text"
          placeholder="Please Enter Name here..."
          value={newUser.name}
          onChange={(e) => setNewUser({ ...newUser, name: e.target.value })}
          required
        />
      </Form.Group>
      <Form.Group>
        <Form.Label>Age</Form.Label>
        <Form.Control
          type="number"
          placeholder="Please Enter Age here..."
          value={newUser.age}
          onChange={(e) => setNewUser({ ...newUser, age: e.target.value })}
          required
        />
      </Form.Group>
      <Form.Group>
        <Form.Label>Address</Form.Label>
        <Form.Control
          type="text"
          placeholder="Please Enter Address here..."
          value={newUser.address}
          onChange={(e) => setNewUser({ ...newUser, address: e.target.value })}
          required
        />
      </Form.Group>
    </Modal.Body>
    <Modal.Footer>
      <Button variant="secondary" onClick={handleClose}>
        Close
      </Button>
      <Button variant="primary" onClick={addUser}>
        Save changes
      </Button>
    </Modal.Footer>
  </Modal>
);

export default AddUserModal;
