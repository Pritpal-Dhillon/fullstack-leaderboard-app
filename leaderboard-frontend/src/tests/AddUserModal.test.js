import React from "react";
import { render, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import AddUserModal from "../Components/AddUserModal";

describe("AddUserModal component", () => {
  const handleCloseMock = jest.fn();
  const addUserMock = jest.fn();
  const setNewUserMock = jest.fn();

  const initialProps = {
    show: true,
    handleClose: handleCloseMock,
    addUser: addUserMock,
    newUser: { name: "", age: "", address: "" },
    setNewUser: setNewUserMock,
  };

  // Clean up
  afterEach(() => {
    jest.clearAllMocks();
  });

  test("renders without crashing", () => {
    const { asFragment } = render(<AddUserModal {...initialProps} />);
    expect(asFragment()).toMatchSnapshot();
  });

  test("input change should call setNewUser", () => {
    const { getByPlaceholderText } = render(<AddUserModal {...initialProps} />);
    const input = getByPlaceholderText("Please Enter Name here..."); 
    fireEvent.change(input, { target: { value: 'New user' } });

    expect(setNewUserMock).toHaveBeenCalledWith({
      name: "New user",
      age: "",
      address: "",
    });
  });

  test('clicking on "Save changes" should call addUser', () => {
    const { getByText } = render(<AddUserModal {...initialProps} />);
    fireEvent.click(getByText("Save changes")); 
    expect(addUserMock).toHaveBeenCalled();
  });

  test('clicking on "Close" should call handleClose', () => {
    const { getByText } = render(<AddUserModal {...initialProps} />);
    fireEvent.click(getByText("Close")); 
    expect(handleCloseMock).toHaveBeenCalled();
  });


});
