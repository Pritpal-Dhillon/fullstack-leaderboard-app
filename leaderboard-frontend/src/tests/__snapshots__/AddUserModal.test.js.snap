// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddUserModal component renders without crashing 1`] = `<DocumentFragment />`;
